velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp

```diff
@@ -647,21 +647,7 @@ class ApproxPercentileAggregate : public exec::Aggregate {
     VELOX_CHECK_EQ(args.size(), 1);
     DecodedVector decoded(*args[0], rows);
     auto rowVec = decoded.base()->as<RowVector>();
-    if constexpr (checkIntermediateInputs) {
-      VELOX_USER_CHECK(rowVec);
-      for (int i = kPercentiles; i <= kAccuracy; ++i) {
-        VELOX_USER_CHECK(rowVec->childAt(i)->isConstantEncoding());
```

Comment:
We need to check all the values are the same because we only use one of them later

Reply:
There's a check in `checkSetPercentile`, I think it's enough now. WDYT

Reply:
BTW, I can't reproduce "Only flat encoding is allowed for percentile array elements", do you have any idea?

Reply:
`checkSetPercentile` is call per batch not per row, we would need something more efficient if we want to check the constraint per row